src/modules/land_detector/MulticopterLandDetector.cpp

```diff
@@ -154,40 +149,88 @@ bool MulticopterLandDetector::_get_ground_contact_state()
-	// throttle-up vibration. If accelerating fast the throttle thresholds will still give
-	// an accurate in-air indication.
 	bool vertical_movement = false;
 
-	if (hrt_elapsed_time(&_landed_time) < LAND_DETECTOR_LAND_PHASE_TIME_US) {
+	if (lpos_available && _vehicle_local_position.v_z_valid) {
+		// Check if we are moving vertically - this might see a spike after arming due to
+		// throttle-up vibration. If accelerating fast the throttle thresholds will still give
+		// an accurate in-air indication.
+		float max_climb_rate = math::min(land_speed_threshold * 0.5f, _param_lndmc_z_vel_max.get());
 
-		// Widen acceptance thresholds for landed state right after arming
-		// so that motor spool-up and other effects do not trigger false negatives.
-		vertical_movement = fabsf(_vehicle_local_position.vz) > _param_lndmc_z_vel_max.get()  * 2.5f;
+		if (hrt_elapsed_time(&_landed_time) < LAND_DETECTOR_LAND_PHASE_TIME_US) {
+			// Widen acceptance thresholds for landed state right after arming
+			// so that motor spool-up and other effects do not trigger false negatives.
+			max_climb_rate = _param_lndmc_z_vel_max.get() * 2.5f;
+		}
 
-	} else {
-		// Adjust max_climb_rate if land_speed is lower than 2x max_climb_rate
-		float max_climb_rate = ((land_speed_threshold * 0.5f) < _param_lndmc_z_vel_max.get()) ? (0.5f * land_speed_threshold) :
-				       _param_lndmc_z_vel_max.get();
-		vertical_movement = fabsf(_vehicle_local_position.vz) > max_climb_rate;
+		vertical_movement = (fabsf(_vehicle_local_position.vz) > max_climb_rate);
 	}
 
+
 	// Check if we are moving horizontally.
-	_horizontal_movement = sqrtf(_vehicle_local_position.vx * _vehicle_local_position.vx
-				     + _vehicle_local_position.vy * _vehicle_local_position.vy) > _param_lndmc_xy_vel_max.get();
+	if (lpos_available && _vehicle_local_position.v_xy_valid) {
+
+		float gndspeed = sqrtf(_vehicle_local_position.vx * _vehicle_local_position.vx
+				       + _vehicle_local_position.vy * _vehicle_local_position.vy);
+
+		_horizontal_movement = (gndspeed > _param_lndmc_xy_vel_max.get());
+
+	} else {
+		_horizontal_movement = false;
+	}
+
 
 	// if we have a valid velocity setpoint and the vehicle is demanded to go down but no vertical movement present,
 	// we then can assume that the vehicle hit ground
-	_in_descend = _is_climb_rate_enabled()
-		      && (_vehicle_local_position_setpoint.vz >= land_speed_threshold);
+	const bool vz_valid = (lpos_available && _vehicle_local_position.v_z_valid);
+
+	if (_vehicle_control_mode.flag_control_climb_rate_enabled && vz_valid) {
+		if (PX4_ISFINITE(_vehicle_local_position_setpoint.vz)) {
+			_in_descend = (_vehicle_local_position_setpoint.vz >= land_speed_threshold);
+		}
+
+	} else {
+		_in_descend = false;
+	}
+
+
+	// low thrust: initially 30% of configured throttle range between min and hover
+	float sys_low_throttle = _params.minThrottle + (_params.hoverThrottle - _params.minThrottle) *
+				 _param_lndmc_low_t_thr.get();
```

Comment:
```suggestion
	// low thrust: configured throttle range between min and hover
	float sys_low_throttle = _params.minThrottle + (_params.hoverThrottle - _params.minThrottle) *
				 _param_lndmc_low_t_thr.get();
```
the 30% is old comment when this was hardcodded